presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -1018,4 +1021,29 @@ public void setPartitionLeases(MetastoreContext metastoreContext, String databas
     {
         throw new PrestoException(NOT_SUPPORTED, "setPartitionLeases is not supported by Glue");
     }
+
+    @Override
+    public long lock(MetastoreContext metastoreContext, String databaseName, String tableName)
+    {
+        long lockid = new Random().nextLong();
+        Map<String, String> properties = new HashMap<>();
+//        properties.put("lock-impl", "org.apache.iceberg.aws.glue.DynamoLockManager");
+        properties.put("lock.table", tableName);
+        properties.put("lock.acquire-timeout-ms", "300000");
+        LockManager lockManager = metastoreContext.getLockManager();
```

Comment:
I have some concerns with the current implementation, this lock method is a common implementation for any table format, and introducing an iceberg dependency doesn't look like an ideal solution. If we look at the implementation for Hive Metastore, it is generic. Can we brainstorm if it would be possible to do something similar in this case as well?

Reply:
In addition to that, this also introduces iceberg dependency in the hive connector.

Reply:
Need to see if a similar LockManager class is available in aws-sdk for java.